packages/library-base/src/commonMain/kotlin/io/realm/RealmConfiguration.kt

```diff
@@ -111,7 +105,7 @@ interface RealmConfiguration {
         // Should always follow Builder constructor arguments
         fun with(
             schema: Set<KClass<out RealmObject>>
-        ): RealmConfiguration {
+        ): LocalConfiguration {
```

Comment:
There is something slightly annoying about returning a `LocalConfiguration` from a `RealmConfiguration.with(...)`, but I guess that is unavoidable. But shouldn't `build()` also return a `LocalConfiguration` then?

Reply:
Yes. `build()` should also return a `LocalConfiguration` 👍 

Reason why I kept the builder in `RealmConfiguration` was just to keep it backward compatible. Further, there will be no scenarios (🙈) where we would like to have a builder on both `RealmConfiguration` and `LocalConfiguration` so I though reusing the `RealmConfiguration` namespace was OK. But maybe we would rather want to make a braking change now and do the clean split now?

Reply:
Yes, I was entertaining the idea...it would also make it really clear about the feature set....but there might be a discoverability issue since it would still be `Realm.open(config: RealmConfiguration)` so it would require som digging to find the correct constructors 🤔

Reply:
Maybe it is fine to keep it as it it now, and then we can think about a breaking change later?

Reply:
I also had a good feeling about the explicit separation of `Local` vs. `Sync`. Since we already have the indirection through the various builders and people cannot just type ahead and construct them anyway, it is maybe ok to just point to the various options of constructing the configurations in the `Realm.open` documentation 🤔

Reply:
Yes, just putting it in the documentation might be enough.

Reply:
I think it would be a reasonable change, but I have left it to stay _non-breaking_ until the idea has settled and maybe piled up with other breaking changes. Added a TODO with link to these comments.

Reply:
To avoid confusion with a new name, I have kept the `RealmConfiguration` for the local configuration but introduced an underlying `Configuration` base interface.